src/bundle/DependencyInjection/Compiler/NavigationHubPass.php

```diff
@@ -24,14 +24,23 @@ public function process(ContainerBuilder $container)
 
     private function processNavigationHubTag(ContainerBuilder $container, $tag, $index)
     {
-        $items = [];
+        $unsorted = [];
         $services = $container->findTaggedServiceIds($tag);
-        foreach (array_keys($services) as $serviceId) {
-            $items[] = new Reference($serviceId);
+        foreach ($services as $serviceId => $tags) {
+            foreach ($tags as $tag) {
+                $priority = isset($tag['priority']) ? $tag['priority'] : 0;
```

Comment:
instead of having a default priority, I think those services (including ours) should always have one to make the whole a process a bit more predictable.